### PR TITLE
Respond to relative positioning via composite

### PR DESCRIPTION
**Pull request by JasonTheKitten**
Relative positioning now uses a composite system. I also geared up a bit for when I add support for fixed and absolute positions.

### CODE REVIEW

File: src/main/java/com/github/webicitybrowser/thready/gui/graphical/base/imp/GUIContentImp.java [hunk 1]
```diff
@@ -174,7 +177,7 @@ private void performRenderPipeline(ScreenContentRedrawContext redrawContext) {
 			performRenderCycle(redrawContext);
 			System.gc();
 		case COMPOSITE:
-			performCompositeCycle();
+			performCompositeCycle(redrawContext);
```
Comment (PiotrJagla) [inline src/main/java/com/github/webicitybrowser/thready/gui/graphical/base/imp/GUIContentImp.java]: Why there arent break or return statements in this switch case? Is it intended that when invalicationLevel is BOX other values are also executed

File: src/main/java/com/github/webicitybrowser/threadyweb/graphical/lookandfeel/weblaf/display/scroll/ScrollCompositor.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.github.webicitybrowser.threadyweb.graphical.lookandfeel.weblaf.display.scroll;
+
+import com.github.webicitybrowser.thready.dimensions.AbsolutePosition;
+import com.github.webicitybrowser.thready.dimensions.Rectangle;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.stage.composite.CompositeLayer.CompositeReference;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.UIPipeline;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.stage.composite.GlobalCompositeContext;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.stage.composite.LocalCompositeContext;
+
+public final class ScrollCompositor {
```
Comment (PiotrJagla) [inline src/main/java/com/github/webicitybrowser/threadyweb/graphical/lookandfeel/weblaf/display/scroll/ScrollCompositor.java:11]: I think that private contructor is missing here

File: src/main/java/com/github/webicitybrowser/threadyweb/graphical/lookandfeel/weblaf/ui/element/styled/StyledUnitCompositor.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.github.webicitybrowser.thready.dimensions.AbsolutePosition;
+import com.github.webicitybrowser.thready.dimensions.Rectangle;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.UIPipeline;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.stage.composite.GlobalCompositeContext;
+import com.github.webicitybrowser.thready.gui.graphical.lookandfeel.core.stage.composite.LocalCompositeContext;
+import com.github.webicitybrowser.threadyweb.graphical.lookandfeel.weblaf.layout.flow.util.BoxOffsetDimensions;
+import com.github.webicitybrowser.threadyweb.graphical.lookandfeel.weblaf.layout.flow.util.BoxPositioningOverride;
+
+public class StyledUnitCompositor {
```
Comment (PiotrJagla) [inline src/main/java/com/github/webicitybrowser/threadyweb/graphical/lookandfeel/weblaf/ui/element/styled/StyledUnitCompositor.java:10]: Also here private contructor is missing and this class is not final

File: src/main/java/com/github/webicitybrowser/thready/gui/graphical/base/imp/GUIContentImp.java (hunk 1 above)
Comment (JasonTheKitten) [reply src/main/java/com/github/webicitybrowser/thready/gui/graphical/base/imp/GUIContentImp.java]: Yes. It is a pipeline, and it performs all levels subsequent to the invalidation level.

**Review by PiotrJagla — APPROVED** [review]
Code approved

**Pull request merged**